### DISCUSSION TITLE
some way to parameterize run files?

### DISCUSSION THREAD

**Discussion by markmarz** [meisnate12/Plex-Meta-Manager#963] [Q&A] [answered]
I'm probably missing something, but I've reviewed the docs repeatedly and just can't find what I need.

I have a separate set of these files (on Debian) for each run:

1. run_xxx.sh: where xxx = name of collection, such as run_mubi.sh. The only thing that changes between each of these files is the named config file, such as **config config/config_mubi.yml**
2. config_xxx.yml = configuration file for collection, such as **config_mubi.sh**. The only thing that changes between each of these files is **- file: config/xxx.yml** under libraries: Movies: metadata_path; such as **- file: config/mubi.yml**
3. xxx.yml = where the templates and collections are defined, such as **mubi.yml**. This is unique for each run.

Surely there must be some way to parameterize this so that a single (1) run and (2) config will suffice, with some sort of command line option to feed in name of configuration for run and pointer to final yml in configuration?

For clarity, here are excepts of each file showing what changes:
**run_mubi.sh:**
  

> docker run --rm  -v "/srv/dev-disk-by-uuid-1c9bf009-c625-4afc-890f-6b278fdf6fe2/dockers/plex-meta-manager:/config:rw" 
>   meisnate12/plex-meta-manager --config **config/config_mubi.yml** --run

**config_mubi.yml:**

> libraries:
>   Movies:                                       # Each library must match the Plex library name
>     metadata_path:
>     - file: **config/mubi.yml**
>     - folder: config/Movies/
> settings:
>   cache: true

**mubi.yml** (this is truly unique and of course needs no parameterization):

> ###################
> external_templates:
> ###################
>   - folder: config/templates
> 
> ############
> collections:
> ############
> 
>   MUBI 1,000 (ranked):
>     url_poster: https://i.imgur.com/c5qLT77.jpg
>     template: { name: ranked_template }
>     trakt_list: https://trakt.tv/users/josuegonzalez/lists/mubi-1000?sort=rank,asc
>     summary: "The greatest films ever made, as voted by MUBI’s global community of film lovers."
> 
>   MUBI 1,000 (alphabetical):

**Comment by meisnate12** [comment, marked as answer]
thats not a thing becasue you can define multiple libraries in a single config file and you can schedule each metadata file to run whenever you want it to

**Reply by markmarz** [reply]
Clearly I don't understand how. Can you point me to where I missed it in the docs and/or give an example here?

**Reply by meisnate12** [reply]
im a little unclear exactly what youre trying to do tbh which is why its hard to point you in the right direction
you can define [multiple libraries](https://metamanager.wiki/en/latest/config/libraries.html#example) like this
```yaml
libraries:
  Movies:
    metadata_path:
      - file: config/Movies.yml
  TV Shows:
    metadata_path:
      - file: config/TV Shows.yml
```
You can schedule a [Metadata File](https://metamanager.wiki/en/latest/config/paths.html#schedule) like so
```yaml
    metadata_path:
      - file: config/TV Shows.yml
        schedule: weekly(saturday)
```
options can be found [here](https://metamanager.wiki/en/latest/metadata/details/schedule.html)

**Reply by markmarz** [reply]
I think you've nailed it, thanks. You've shown how I can invoke multiple targets with a single config file named in a single run. That's exactly what I missed.

Thanks!
